core/core.go

```diff
@@ -854,16 +855,24 @@ func (t *TextileNode) GetPhotos(offsetId string, limit int, album string) *Photo
 	// query for available hashes
 	a := t.Datastore.Albums().GetAlbumByName(album)
 	if a == nil {
-		return &PhotoList{Hashes: make([]string, 0)}
+		return &PhotoList{
+			Hashes: make([]string, 0),
+			Paths:  make([]string, 0),
+		}
 	}
 	list := t.Datastore.Photos().GetPhotos(offsetId, limit, "album='"+a.Id+"'")
 
 	// return json list of hashes
 	res := &PhotoList{
 		Hashes: make([]string, len(list)),
+		Paths:  make([]string, len(list)),
 	}
 	for i := range list {
-		res.Hashes[i] = list[i].Cid
+		password := ksuid.New().String()
+		hash := list[i].Cid
+		t.HashPasses[hash] = password
+		res.Hashes[i] = hash
+		res.Paths[i] = hash + ":" + password + "@localhost" + t.GatewayProxy.Addr + "/ipfs/" + hash
```

Comment:
do we need to pass `hash` three times?

Reply:
ok cool. if you need the pieces of the path client side, then i'd just pass the pieces... 
```
type PhotoListItem struct {
	Hash  string `json:"hash"`
	Token string `json:"token"`
	Proto string `json:"proto"`
	Host  string `json:"host"`
}

type PhotoList struct {
	Items []PhotoListItem `json:"items"`
}
```

the idea wiht passing the whole path, was to make RN side easier, not harder! :)

Reply:
let's just move on though, easy change later...